input/cql/HIVElements.cql

```diff
@@ -804,6 +789,29 @@ define "Date informed of HIV-positive diagnosis Value":
   return O.value
 /* End of Date informed of HIV-positive diagnosis */
 
+/*
+@activity: HIV.B7 Test for HIV using testing algorithm
+*/
+define "HIV test":
+  [ServiceRequest: Concepts."HIV test type - HIV.B.DE81 Choices"] SR
+  where SR.status in { 'active', 'completed' }
+
+define "Non-Self HIV test":
+  "HIV test" SR
+  where SR.code !~ Concepts."HIV self-test"
+
+define "HIV test result DiagnosticReport":
+  [DiagnosticReport: Concepts."HIV test type - HIV.B.DE81 Choices"] DR
+  with "HIV test" SR
+    such that DR.basedOn.references(SR)
+  where DR.status in { 'final', 'amended', 'corrected' }
```

Comment:
Do we really want to make this dependent on a DiagnosticReport? In this case, it's not clear to me what value the DR adds over the Observation for our purposes (i.e., the Observation still has the `basedOn` stuff, we're only really checking a single observation, etc.